lua/pug/modules/lagdetection.lua

```diff
@@ -52,13 +52,27 @@ assert( type( lag.fPanic ) == "function", "Invalid PanicMethod variable" )
 	Compare new rates against the average. (Test ZScore vs Subtraction.)
 ]]
 
+
 local function getMean()
+    --local count = 0
+    --for _, val in pairs(sample.data) do
+    --    count = count + val
+    --end
+	--return count / table.Count(sample.data)
+
 	local x = tableReduce( function(a, b) return a + b end, sample.data )
 	return ( x / #sample.data )
 end
 
+local maxRate = 1 / engine.TickInterval()
+local sampleTime = 1.5 -- time in seconds to average
+local sampleSize = sampleTime * maxRate
```

Comment:
This was an easier way for us to say "keep track of roughly the last X seconds of data"

We found that 1.5 was a good number. When trying 3 we noticed some weird behavior.

Reply:
This can be very problematic as some servers experience mild to major lag on start up, thus returning incorrect or inconsistent values. As this gets the current "true" tick rate of the server, and not the rate the server is set to.